host_core/lib/host_core/host.ex

```diff
@@ -73,13 +73,102 @@ defmodule HostCore.Host do
 
     publish_host_started(labels, friendly_name)
 
-    {:ok,
-     %State{
-       host_key: opts.host_key,
-       lattice_prefix: opts.lattice_prefix,
-       friendly_name: friendly_name,
-       labels: labels
-     }}
+    state = %State{
+      host_key: opts.host_key,
+      lattice_prefix: opts.lattice_prefix,
+      friendly_name: friendly_name,
+      labels: labels
+    }
+
+    if config_service_enabled?(opts) do
+      {:ok, state, {:continue, :load_supp_config}}
+    else
+      {:ok, state}
+    end
+  end
+
+  # truthy
+  defp config_service_enabled?(opts) do
+    String.upcase(opts.config_service_enabled) in [
```

Comment:
so many suggestions for this but in the interest in keeping this function under 100 lines I'll hold back